frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -391,6 +393,37 @@ def circuit(x):
         _, observed_shape = tree_flatten(observed)
         assert expected_shape == observed_shape
 
+    def test_mcm_method_with_zne(self, backend):
+        """Test that the dynamic_one_shot works with ZNE."""
+        dev = qml.device(backend, wires=1, shots=5)
+
+        def circuit():
+            return qml.expval(qml.PauliZ(0))
+
+        mcm_method = "one-shot"
+
+        @qjit
+        def mitigated_circuit():
+            s = jnp.array([1, 2])
+            g = qml.QNode(circuit, dev, mcm_method=mcm_method)
+            return mitigate_with_zne(g, scale_factors=s)()
+
+        observed = mitigated_circuit()
+        mcm_method = None
+        expected = mitigated_circuit()
+
+        assert np.allclose(expected, observed)
+
+    def test_get_call_jaxpr(self):
+        """Test _get_call_jaxpr raises AsserionError if no function primitive exists."""
+
+        def f(x):
+            return x * x
+
+        jaxpr = make_jaxpr(f)(2.0)
+        with pytest.raises(AssertionError, match="No call_jaxpr found in the JAXPR"):
+            _ = _get_call_jaxpr(jaxpr)
```

Comment:
Since this test is not related to mcms, could you move it `pytest/test_jax_primitives.py` (creating it if it doesn't exist) ?